draft-ietf-core-dns-over-coap.md

```diff
@@ -551,28 +555,31 @@ DNS Push Notifications {{-dns-push}} provides the capability to asynchronously n
 However, it results in additional overhead, which conflicts with constrained resources.
 This is the reason why it is RECOMMENDED to use CoAP Observe {{-coap-observe}} instead of DNS Push
 in the DoC domain.
+This is particularly useful if a short-lived record is requested frequently.
 The DoC server SHOULD provide Observe capabilities on its DoC resource and do as follows.
 
-A DoC server
-MAY subscribe to DNS push notifications,
-which involves sending a DNS Subscribe message (see ({{Section 6.2 of -dns-push}}),
+If a DoC clients wants to observe a resource record, a DoC server acts as subject and
+adds the client to its list of observers for that resource in accordance to {{-coap-observe}}.
+The DoC server MAY subscribe to DNS push notifications for that record.
+This involves sending a DNS Subscribe message (see ({{Section 6.2 of -dns-push}}),
 instead of a classic DNS query to fetch the
-information on behalf of a DoC client.
-This is particularly useful
-when a CoAP request indicates that the DoC client wants to observe a resource record,
-or when a short-lived record is requested frequently.
+information on behalf of the DoC client.
+
 After the list of observers for a particular DNS query has become empty
 (by explicit or implicit cancellation of the observation as per {{Section 3.6 of -coap-observe}}),
 and no other reason to subscribe to that request is present,
 the DoC server SHOULD cancel the corresponding subscription.
 This can involve sending an DNS Unsubscribe message or closing the session (see {{Section 6.4 of -dns-push}}).
+As there is no CoAP observer anymore from the perspective of the DoC server, a failure to do so cannot be communicated back to any DoC observer.
+As such, error handling (if any) needs to be resolved between the DoC server and the upstream DNS infrastructure.
 
 Whenever the DoC server receives a DNS Push message from the DNS
 infrastructure for an observed resource record, the DoC server sends an appropriate Observe notification response
 to the DoC client.
 
-A server that cannot proxy to DNS Push (or any other means of obtaining new records) should not send the response as a notification
-(i.e., it should not send the Observe option).
+A server that cannot proxy to DNS Push (or any other means of obtaining new records) SHOULD respond with a plain response message (i.e., without the Observe option) to a registration attempt.
+Observe notification for other purposes, however, are still possible.
+For instance, if the DoC server wants to notify about a change to a record it is the authoritative name server for, e.g., a new IP address for the name.
```

Comment:
I think this'd cause more confusion than being helpful, esp. as to what the other purposes are -- and we're more in the factual requirements space here than in the normative SHOULD.

Maybe:

```suggestion
A server that responds with notifications (i.e., with the Observe option)
needs to have a means of obtaining current representations.
This may happend through DNS Push,
but also by polling
or implicitly (if the DoC is the authoritative name server for the request).
```

Reply:
This suggestion needs some fixes IMHO.

```suggestion
A server that responds with notifications (i.e., a response with the Observe option)
needs to have means of obtaining current resource records.
This may happen through DNS Push as described above,
but also by upstream polling
or through implicit circumstances (e.g., if the DoC is the authoritative name server for the record and wants to notify about changes).
```

Reply:
Maybe a bit less verbose?

```suggestion
A server that responds with notifications (i.e., sends the Observe option)
needs to have means of obtaining current resource records.
This may happen through DNS Push,
but also by upstream polling
or through implicit circumstances (e.g., if the DoC is the authoritative name server for the request).
```

Reply:
Then let's do a bit less verbose, but also a bit more precise version ;-)

```suggestion
A server that responds with notifications (i.e., sends the Observe option) needs to have means of obtaining current resource records.
This may happen through DNS Push, but also by upstream polling or implicit circumstances (e.g., if the DoC server is the authoritative name server for the record and wants to notify about changes).
```